ledger/ledger-offset/src/test/suite/scala/com/daml/ledger/offset/HexOffsetSpec.scala

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.offset
+
+import com.daml.lf.data.Ref
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks.forAll
+import org.scalatest.prop.Tables.Table
+import org.scalatest.wordspec.AnyWordSpec
+
+class HexOffsetSpec extends AnyWordSpec with Matchers {
+
+  "building lexicographical string" should {
+
+    "return none for no valid option" in {
+      HexOffset.firstBefore(
+        Ref.HexString.assertFromString("000000000000")
+      ) shouldBe None
+    }
+
+    "return expected lower offsets" in {
+      forAll(
+        Table(
+          "offset" -> "expected offset",
+          "00000001" -> "00000000",
+          "0000000a" -> "00000009",
+          "00000100" -> "000000ff",
+          "a0000000" -> "9fffffff",
+          "00000ab0" -> "00000aaf",
+          "00007a90" -> "00007a8f",
+          "00007a94" -> "00007a93",
```

Comment:
Does it make sense to try different lengths? How about property-based?

Reply:
It shouldn't make a difference when it comes to length.
I was thinking of using property-based testing using scalacheck, but we should decide if we want to use that everywhere and add support by creating generators for all the LF structures.

Reply:
I had the impression that I had to add the scalacheck deps and set it up, but it's already there, therefore I added a property-based test case.